### PR TITLE
[EFR32] Make sure Logs cannot write out of bounds, cleanup examples and build system.

### LINKED ISSUE #4732: EFR32 logging code can write buffers out of bounds
#### Problem
`src/platform/EFR32/Logging.cpp` has this code:
```
        constexpr size_t maxPrefixLen = ChipLoggingModuleNameLen + 3;
        constexpr size_t maxPrefixLen = chip::Logging::kMaxModuleNameLen + 3;
        static_assert(sizeof(formattedMsg) > maxPrefixLen);
```
and then proceeds to use `strcpy` to copy things like `"<detail> "` into `formattedMsg`.  That's longer than `maxPrefixLen` as defined above (which ends up being `6`), so if `CHIP_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE` is small enough this will write outside the stack buffer.

 #### Proposed Solution
Fix the length check to actually account for the things we plan to put into this buffer.

Note: https://github.com/project-chip/connectedhomeip/pull/4627 touched this code, but just to rename variables; the problem predates that PR.

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
### Problem
EFR32 logging code can potentially write buffers out of bounds. 
Missing CATEGORY prefix in the logs.
Undefining EFR32_LOG_ENABLED leaves some functions and variables that are never used.
Some files are duplicated for each boards init but are not even used when the example use Thread.
Some Thread files are built for examples that do not use thread.

#### Fix EFR Logs
- Validate buffer size is large enough to contain the log Category and module name prefixes.
- Re-add the `CATEGORY` prefix in the logs.
- When EFR32_LOG_ENABLED is not defined, removes all EFR log functions and variables.

#### Clean up EFR platform init and rework the example GN Build order:
- Enable Openthread only for the needed example.
	- Set chip_enable_openthread to true in args.gni of the desired examples.
	- Thread is not used in persitent-storage and pigweed-app examples.
	- We also plan to have a WiFi example.
- Remove init_mcu and init_board for each EFR board supported.
- Rename init_otSystem.c to init_efrPlatform.cpp to take care of the platform initialization for any specified board and ifdef Thread related initialization.
- Resize BLE APP task stack size. The stack was almost fully used and adding some code caused a stack overflow.
- Some import sequence in gn files were a bit wonky. Re-organized it.

 Fixes #4732, #5107

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/platform/EFR32/Logging.cpp [hunk 1]
```diff
@@ -59,19 +59,22 @@
 #define LOG_DETAIL "<detail> "
 #define LOG_LWIP "<lwip  > "
 #define LOG_EFR32 "<efr32 > "
+static constexpr uint8_t kMaxCategoryStrLen = 9;
```
Comment (bzbarsky-apple) [inline src/platform/EFR32/Logging.cpp:62]: Please document that this needs to be long enough to handle any of the LOG_* things (not including their null-terminator)?

Or better yet would be to compute it as https://github.com/project-chip/connectedhomeip/issues/4732#issuecomment-775393921 suggests, so it's obvious what's going on.  I just put up https://github.com/project-chip/connectedhomeip/pull/5124 to implement such a thing.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@jmartinez-silabs - pullaprove complains about CLA signing. Could you sign so we can merge?

**Comment by andy31415** [comment]
Tried to move to draft and back to see if that convinces CLA bot too run ... but it does not seem so.

**Comment by jmartinez-silabs** [comment]
Rebasing with master to trigger checks again

File: src/platform/EFR32/Logging.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/platform/EFR32/Logging.cpp:62]: I will update this seperatly with what you did in #5124.

**Pull request merged**